### PR TITLE
update README

### PR DESCRIPTION
**Pull request by koriym**
- add Overview and Getting start example.
- more detail for `toConstructor` binding
- Remove related only version 1.x

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,65 +8,139 @@
...
+## Creating Object graph 
+
+With dependency injection, objects accept dependencies in their constructors. To construct an object, you first build its dependencies. But to build each dependency, you need its dependencies, and so on. So when you build an object, you really need to build an object graph.
+
+Building object graphs by hand is labour intensive, error prone, and makes testing difficult. Instead, Ray.Di can build the object graph for you. But first, Ray.Di needs to be configured to build the graph exactly as you want it.
+
+To illustrate, we'll start the MovieLisetr class that accepts its dependent interfaces MovierFinderInterface in its constructor.
 
-class Finder implements FinderInterface
+```
```
Comment (malukenho) [inline README.md]: php

File: README.md [hunk 2]
```diff
@@ -8,65 +8,139 @@
...
-        $this->finder = $finder;
+        $this->bind(ProcessorInterface::class)->to(PaypalProcessor::class); 
+        $this->bind(LoggerInterface::class)->to(DatabaseLogger::class);
     }
 }
+```
+
+The modules are the building blocks of an injector, which is Ray.Di's object-graph builder. First we create the injector, and then we can use that to build the BillingService:
+
+```
```
Comment (malukenho) [inline README.md]: php

Comment (koriym) [inline README.md]: @malukenho Thanks !

**Comment by koriym** [comment]
@malukenho I would like to improve documentation. Any feedback is welcome for this new version of README. Any uncertain things ?
https://github.com/koriym/Ray.Di/blob/c6f62798c1a37c1110e77a568a68926baa3a14d7/README.md

**Pull request merged**